cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -517,6 +527,17 @@ impl<I: VCodeInst> VCode<I> {
                 cur_srcloc = None;
             }
 
+            // Emit the constants used by the function.
+            let used_constants = buffer
+                .used_constants()
+                .map(|(c, l)| (*c, *l))
+                .collect::<Vec<_>>();
+            for (used_constant, label) in used_constants {
+                let data = self.used_constants.retrieve_use(used_constant);
+                // TODO figure out how we want to align these.
+                buffer.defer_constant(label, 16, data.as_slice(), u32::max_value());
```

Comment:
Would it be sufficient in all cases to align to the size (so e.g. 8-byte constants are 8-byte-aligned, ...)? Or are there cases where we need more than that?

Reply:
Check out the new `VCodeConstantData::alignment` and let me know what you think.